test/parallel/test-net-dns-custom-lookup.js

```diff
@@ -0,0 +1,37 @@
+var common = require('../common');
+var assert = require('assert');
+var net = require('net');
+var dns = require('dns');
+var ok = false;
+
+function check(addressType, cb) {
+  var server = net.createServer(function(client) {
+    client.end();
+    server.close();
+    cb && cb();
+  });
+
+  var address = addressType === 4 ? '127.0.0.1' : '::1';
+  server.listen(common.PORT, address, function() {
+    net.connect({
+      port: common.PORT,
+      host: 'localhost',
```

Comment:
This was actually something I am not sure of. Since we don't call lookup in the event that the host is an IP address, this test could fail of `common.localhostIPv4 === '127.0.0.1'`